contracts/TokenVesting.sol

```diff
@@ -2,355 +2,442 @@
 
     /**
-    * @dev Returns the amount of tokens that can be withdrawn by the owner.
-    * @return the amount of tokens
-    */
-    function getWithdrawableAmount()
-        public
-        view
-        returns(uint256){
-        return _token.balanceOf(address(this)).sub(vestingSchedulesTotalAmount);
+     * @notice Batch create vesting schedules.
+     * @param _vestingSchedules vesting schedules to create
+     */
+    function batchCreateVestingSchedule(
+        VestingSchedule[] calldata _vestingSchedules
+    ) external onlyOwner {
+        unchecked {
+            uint256 _vestingSchedulesTotalAmount = vestingSchedulesTotalAmount;
+            uint256 totalAmount = _vestingSchedulesTotalAmount;
+            uint256 length = _vestingSchedules.length;
+            for (uint256 i = 0; i < length; i++) {
+                VestingSchedule calldata vestingSchedule = _vestingSchedules[i];
+                address beneficiary = vestingSchedule.beneficiary;
+                if (beneficiary == address(0)) revert InvalidBeneficiary();
+                uint112 amount = vestingSchedule.amountTotal;
+                if (amount == 0) revert InvalidAmount();
+                totalAmount += amount;
+                uint32 start = vestingSchedule.start;
+                if (vestingSchedule.cliff < start) revert CliffBeforeStart();
+                uint32 duration = vestingSchedule.duration;
+                if (duration == 0) revert InvalidDuration();
+                require(start + duration > start);
+                bytes32 vestingScheduleId = computeNextVestingScheduleIdForHolder(
+                        beneficiary
+                    );
+                vestingSchedules[vestingScheduleId] = vestingSchedule;
+                vestingSchedulesIds.push(vestingScheduleId);
+                ++holdersVestingCount[beneficiary];
+                emit Created(vestingScheduleId, beneficiary, amount);
+            }
+            require(totalAmount > _vestingSchedulesTotalAmount);
+            vestingSchedulesTotalAmount = totalAmount;
+        }
     }
 
     /**
-    * @dev Computes the next vesting schedule identifier for a given holder address.
-    */
-    function computeNextVestingScheduleIdForHolder(address holder)
-        public
-        view
-        returns(bytes32){
-        return computeVestingScheduleIdForAddressAndIndex(holder, holdersVestingCount[holder]);
+     * @notice Creates a new vesting schedule for a beneficiary.
+     * @param _beneficiary address of the beneficiary to whom vested tokens are transferred
+     * @param _start start time of the vesting period
+     * @param _cliff duration in seconds of the cliff in which tokens will begin to vest
+     * @param _duration duration in seconds of the period in which the tokens will vest
+     * @param _revocable whether the vesting is revocable or not
+     * @param _amount total amount of tokens to be released at the end of the vesting
+     */
+    function createVestingSchedule(
```

Comment:
Why not call batch create with a single `VestingSchedule`? (would need some refactoring for sure to make `onlyOwner` work).

Reply:
How do you create `VestingSchedule[] calldata` from a single `VestingSchedule`?